### DISCUSSION TITLE
BT-01(f)-Procedure mandatory in the fields.json but not on the eNotice2 preview site

### DISCUSSION THREAD

**Discussion by DeanofHouseHanson** [OP-TED/eForms-SDK#943] [Q&A] [answered]
Hi All,

I've been working on an integration and when comparing the fields I have as mandatory against the eNotice2 site I have noticed that BT-01(f)-Procedure is mandatory for me when it shouldn't be.

In my example, I'm using notice type 29. This field in the fields.json file is flagged as non-mandatory but in the constraints array, it holds all of the notice types and a value of true for mandatory:
![image](https://github.com/OP-TED/eForms-SDK/assets/122117628/9c7fac0a-f268-43a9-9617-24428e98745f)

According to the documentation here:
![image](https://github.com/OP-TED/eForms-SDK/assets/122117628/cbf850a0-876e-447c-893e-ab1cf2cdc44f)

We should be setting this to mandatory as notice type 29 exists in the noticeTypes array.

When reviewing this on the eNotice2 website you see it as non-mandatory.
![image](https://github.com/OP-TED/eForms-SDK/assets/122117628/a8aae3fc-58d5-41a9-b5e9-2d765015110d)

Anyone else encountered this kind of issue? And if so how have you resolved it?
Thanks!

**Comment by bertrand-lorentz** [comment, marked as answer]
In the documentation page from your screenshot, there is the following note, indicated with a "CAUTION":

"The value of the forbidden property must take precedence over the value the mandatory property. If a field is forbidden, then it should not be present in the notice regardless of the value of its mandatory property. You should always check first if a field is forbidden or not. Then, consider whether the field is mandatory or optional only if the field is not forbidden."

In your case, for notice subtype 29, the field BT-01(f)-Procedure is forbidden when BT-01(e)-Procedure is not present.
When it is not forbidden (so when BT-01(e)-Procedure is present), it is then mandatory.

**Reply by douret** [reply]
Hello @bertrand-lorentz,

I do not understand your answer.

BT-01(e)-Procedure is a hidden field so it present.
So if i understand correctly your answer, field BT-01(f)-Procedure has to be mandatory which is not the case in TED.

Please see also my question about the same subject: https://github.com/OP-TED/eForms-SDK/discussions/1202

**Comment by douret** [comment]
@bertrand-lorentz ,  @pdonohoe 

Could you please provide us a feed back on this issue and https://github.com/OP-TED/eForms-SDK/discussions/1202
We do not understand your answers.

Regards

**Comment by pdonohoe** [comment]
You said above: "BT-01(e)-Procedure is a hidden field so it present." This is not correct.

The XML element for BT-01(e)-Procedure is cbc:ID within cac:ProcurementLegislationDocumentReference, which can be used for all three legal bases. BT-01(e)-Procedure is used to state which legal basis is being used. In the case of Cross-Border Law, or Local Legal Basis where there is no identifier, cbc:ID contains a fixed text: either "CrossBorderLaw" or "LocalLegalBasis".

eNotices2 is designed to make it easier to fill notices in. The selections for Cross-Border Law and Local Legal Basis are in different places on the screen. As they are separated, and the selection of one automatically selects which legal basis type is meant, there is no need to provide a box to specify BT-01(e)-Procedure or BT-09(a)-Procedure. If you don't put something in the box for BT-01(f)-Procedure, then eNotices2 does not add BT-01(e)-Procedure. If you do put something in the box for BT-01(f)-Procedure, then eNotices2 adds BT-01(e)-Procedure.
